bpf/lib/wireguard.h

```diff
@@ -113,31 +116,30 @@ wg_maybe_redirect_to_encrypt(struct __ctx_buff *ctx)
 		goto encrypt;
 #endif /* TUNNEL_MODE */
 
+#ifndef ENABLE_NODE_ENCRYPTION
+	/* A pkt coming from L7 proxy (i.e., Envoy on behalf of a client pod)
+	 * has src IP addr of a host, but not of the client pod (if
+	 * --dnsproxy-enable-transparent-mode=false). Such a pkt must be
```

Comment:
Nit. There seems to be a bit of mismatch in the comment, as it refers to Envoy, but then mentions the `dnsproxy-enable-transparent-mode` which applies to the DNS proxy. Maybe it could be `... (i.e., Envoy or the DNS proxy on behalf ...`.

Reply:
I used "L7 proxy" to refer to both (